lib/solver_dispatcher.ml

```diff
@@ -0,0 +1,14 @@
+(* TODO: put this in some other more appropriate module? *)
+type solver_type =
+  | Z3_solver
+  | Z3_solver2
+  | Cvc5_solver
+  | Colibri2_solver
+  | Bitwuzla_solver
+
```

Comment:
What about providing a `val solver_type_of_string : string -> (solver_type, string) Result.t` along the way ?

Something like:

```ocaml
let solver_type_of_string s =
  match String.map Char.lowercase_ascii s with
  | "z3" -> Ok Z3_solver
  | "z32" -> Ok Z3_solver2 (* I don't like this name TBH, make we could make the intent more clear ? *)
  | "colibri2" -> Ok Colibri2
  | "bitwuzla" -> Ok Bitwuzla_solver
  | "cvc5" -> Ok Cvc5_solver
  | s -> Format.ksprintf Result.error "unknown solver %s" s
```

Reply:
Seems like a good idea to me. 

>  | "z32" -> Ok Z3_solver2 (* I don't like this name TBH, make we could make the intent more clear ? *)

I'll just remove `Z3_solver2` for the moment. It's to experimental, and probably buggy too